pages/api/v1/answers.ts

```diff
@@ -457,6 +458,67 @@ async function Angela( // Good ol' Angela
     );
   }
 
+  const model = org.model;
+  const userQuery = nonSystemMessages[nonSystemMessages.length - 1].content;
+
+  actions = await filterActions(actions, conversationId, userQuery, model);
+
+  let numOpenAIRequests = 0;
+  let totalCost = 0;
+
+  if (actions.length === 0) {
+    console.log(
+      "No actions are relevant to the user's request, using simple chat model",
+    );
+    const prompt = simpleChatPrompt(
+      userQuery,
+      reqData.user_description,
+      org,
+      language,
+    );
```

Comment:
I don't really get the reason for this

It ignores all the chat history (e.g. if this happens midway through a conversation) - surely this is bad? Also, a lot of code below here is repeated further down

Couldn't we just use this prompt format in `getMessages()` if `actions === []`? Then it'll go through basically the same flow of code with knowledge of past messages (saving code and improving the responses)?

Reply:
yep nice